talk/xmpp/contentwrapper.cc

```diff
@@ -0,0 +1,214 @@
+#include "contentwrapper.h"
+
+#include <stdlib.h>
+
+namespace buzz{
+
+  ContentWrapper::ContentWrapper(const std::string& host, const std::string& lang): host(host), lang(lang){};
+
+  ContentWrapper::~ContentWrapper(){};
+
+  std::string ContentWrapper::GenerateLoginStart()
+  {
+    std::string start_string = "<stream:stream to=\'" + host + "\' ";
+    start_string += "xml:lang=\'" + lang + "\' ";
+    start_string += "version=\'1.0\' ";
+    start_string += "xmlns:stream=\'http://etherx.jabber.org/streams\' ";
+    start_string += "xmlns=\'jabber:client\'>\r\n";
+    return start_string;
+  }
+
+  std::string ContentWrapper::GenerateLoginRestart()
+  {
+    return GenerateLoginStart();
+  }
+
+  std::string ContentWrapper::GenerateLogout()
+  {
+    return "</stream:stream>";
+  }
+
+  std::string ContentWrapper::GenerateRequest(const std::string& xml)
+  {
+    return xml;
+  }
+
+  bool ContentWrapper::HandleRecvData(std::string& xml, std::size_t len)
+  {
+    xml = xml.substr(0, len);
+    return true;
+  }
+
+  BoshContentWrapper::BoshContentWrapper(const std::string& host, const std::string& lang, const std::string& server, int port) :
+    ContentWrapper(host, lang),
+    server(server),
+    port(port),
+    rid(0),
+    sid(""),
+    path("/http-bind/"),
+    start_sent(false)
+  {
+    bosh_host = host + ":" + std::to_string((long)port);
+  }
+
+  BoshContentWrapper::~BoshContentWrapper()
+  {
+  }
+
+  std::string BoshContentWrapper::GenerateLoginStart()
+  {
+    start_sent = true;
+    std::string start_string = GetStartString();
+    return WrapHtmlRequest(start_string);
+  }
+
+  std::string BoshContentWrapper::GenerateLoginRestart()
+  {
+    start_sent = true;
+    std::string restart_string = GetRestartString();
+    return WrapHtmlRequest(restart_string); 
+  }
+
+  std::string BoshContentWrapper::GenerateLogout()
+  {
+    std::string logout_string = GetLogoutString();
+    return WrapHtmlRequest(logout_string);
+  }
+
+  std::string BoshContentWrapper::GenerateRequest(const std::string& xml)
+  {
+    std::string request = WrapBodyTag(xml);
+    return WrapHtmlRequest(request);
+  }
+
+  std::string BoshContentWrapper::GetLogoutString()
+  {
+    ++rid;
+    std::string requestbody = "<body";
+    requestbody += "rid=\'" + std::to_string((long)rid) + "\' ";
+    requestbody += "sid=\'" + sid + "\' ";
+    requestbody += "type=\'terminate\'";
+    requestbody += "xmlns=\'" + kXmlnsHttpbind + "\'/>";
+    rid = 0;
+    sid = "";
+    start_sent = false;
+    return requestbody;
+  }
+
+  std::string BoshContentWrapper::GetStartString()
+  {
+    rid = rand() % 100000 + 1728679472;
```

Comment:
Why this? Seems like a 'magic' number (the 1728679472) and I don't believe this is speced